src/components/RegisterCard.vue

```diff
@@ -0,0 +1,131 @@
+<template>
+    <section class="container login-register__wrapper">
+        <div class="header__wrapper">
+            <h1 class="header__big header--black">Ahoy you!</h1>
+            <h1 class="header__big header--grey">Care to register?</h1>
+        </div>
+        <form @submit.prevent="onSubmit" novalidate>
+
+            <p :class="{'red': !!emailError}"></p>
+
+            <CustomInput 
+                v-model="email"
+                label="email"
+                placeholder="Something ending with monterail.com"
+                type="email"
+                @blur="emailTouched = true"/>
+
+        
+
+            <PasswordInput
+                v-model="password"
+                label="password"
+                placeholder="Enter your password"
+                @blur="passwordTouched = true"
+                />
+
+                <p> {{ emailError }}</p>
+
+                <p>{{ passwordError }}</p>
+
+                <p :class="{red: !passwordLength && passwordTouched}">At least 8 characters</p>
+
+                <p :class="{red: !passwordLetters && passwordTouched}">At least one letter</p>
+
+                <p :class="{red: !passwordDigits && passwordTouched}">At least one digit</p>
+
+            <router-link to="/login" class="header__link">Log in instead</router-link>
+
+            <button class="button button--small"
+            type="submit" :disabled="!isFormValid">Next step</button>
+
+        </form>
+    </section>
+</template>
+
+<script>
+import CustomInput from "@/components/CustomInput.vue";
+import PasswordInput from "@/components/PasswordInput.vue";
+
+
+export default {
+    components: {
+        CustomInput,
+        PasswordInput
+    },
+    data() {
+        return {
+            name: 'name',
+            email: '',
+            password: '',
+            emailTouched: false,
+            passwordTouched: false
+        }
+    },
+    methods: {
+        onSubmit() {
+            this.touchAll();
+            if (this.isFormValid) {
+                console.log(`Email: ${this.email}, Password: ${this.password}`)
+            }
+        },
+        touchAll() {
+            this.emailTouched = true;
+            this.passwordTouched = true;
+        },
+        isEmailValid(email) {
+            let emailRegExp =/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+            return email && emailRegExp.exec(email);
+        },
+        isPasswordValid(password) {
+            return (
+            password && this.passwordLength && this.passwordLetters && this.passwordDigits
```

Comment:
so e.g. `isPasswordValid('nodigits')` may be valid 🤔 

I mean that different parts of the expression are checking different password:

```
password && this.passwordLength && this.passwordLetters && this.passwordDigits
    ^                ^                       ^                        ^
  method
 argument           data                    data                     data
```

(my suggestion would be to make it a computed value instead of a method)